File: C++/9. Palindrome Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        int xtemp=x;
        int power=1;
        if(x>INT_MAX || x<0)
        {
            return false;
        }
        if(x<10)
        {
            return true;
        }
        while(xtemp/10>0)
        {
            power *=10; 
            xtemp/=10;
        }
        if(power==1)
        {
            return true;
        }
        while(power>1)
        {
            if( x%10 != x/power )
            {
                return false;
            }
            else
            {
                x=x%power/10;
                power/=100;
            }
        }
        return true;
    }
};

File: python/9. Palindrome Number.py
class Solution:
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0: return False
        power = 1
        y = x
        while y >= 10:
            y //= 10
            power *= 10
        while power :
            if x // power != x % 10:
                return False
            else:
                x = (x % power // 10)
                power //= 100
        return True
